### ISSUE TITLE
When multiple locales (in Next.js) getPathsFromContext() creates 404 paths for untranslated (in Drupal) nodes.

### ISSUE THREAD

**Issue by theRuslan** [chapter-three/next-drupal#147]
> From https://next-drupal.org/docs/reference#getpathsfromcontext:
> "If you have [locales configured](https://nextjs.org/docs/advanced-features/i18n-routing) in your `next.config.js` file, it will automatically create localized routes."

When multiple locales (in Next.js) getPathsFromContext() creates 404 paths for untranslated (in Drupal) nodes.

**Comment by shadcn** [comment]
Can you share the `next.config.js`?

**Comment by TheRuslan** [comment]
Yes. And I also use next-i18next for i18n. Here is both configs.

next.config.js:
```
/** @type {import('next').NextConfig} */

const withPlugins = require("next-compose-plugins");
const { i18n } = require("./next-i18next.config");
const withPWA = require("next-pwa");

const nextConfig = {
  publicRuntimeConfig: {
    canonicalDomain: "https://greenwise.ru",
  },
  reactStrictMode: true,
  images: {
    domains: [process.env.NEXT_IMAGE_DOMAIN],
  },
  i18n,
  experimental: {
    outputStandalone: true,
  },
};

module.exports = withPlugins([
  [
    withPWA,
    {
      pwa: {
        dest: "public",
        disable: process.env.NODE_ENV === "development",
        register: true,
        sw: "/sw.js",
      },
    },
  ],
  [nextConfig],
]);
```

next-i18next.config.js:
```
const path = require("path");

module.exports = {
  i18n: {
    defaultLocale: "ru",
    locales: ["ru", "en"],
    localeDetection: false,
  },
  localePath: path.resolve("locales"),
};

```

**Comment by TheRuslan** [comment]
I will try to reproduce this issue on clean next/drupal setup with the same language configuration.

**Comment by shadcn** [comment]
@TheRuslan what does the pattern (path alias) for the node type looks like?

You should use `[node:source:title]` so that Next.js can automatically create the paths.

![path-alias-page](https://user-images.githubusercontent.com/124599/162932224-c96ab816-41f4-45f4-9a40-9adb2aef95ce.jpg)

**Comment by TheRuslan** [comment]
@shadcn I used [node:title] and tried [node:source:title]. I think in this case it doesn't matter.

I made an example with 2 nodes:
- **1 product** which is **translated** node with the **russian original** and **english translation**
- **1 blog post** which is **untranslated** node with the **russian original only**

In accordance with next-drupal documentation here https://next-drupal.org/docs/reference#getpathsfromcontext
> If you have [locales configured](https://nextjs.org/docs/advanced-features/i18n-routing) in your `next.config.js` file, it will automatically create localized routes.

So getPathsFromContext returns such result:
```
[
    {
        "params": {
            "slug": [
                "blog",
                "test-article-original-node-russian-without-translation"
            ]
        },
        "locale": "ru"
    },
    {
        "params": {
            "slug": [
                "blog",
                "test-article-original-node-russian-without-translation"
            ]
        },
        "locale": "en"
    },
    {
        "params": {
            "slug": [
                "product",
                "testovyy-produkt-original-node-russian"
            ]
        },
        "locale": "ru"
    },
    {
        "params": {
            "slug": [
                "product",
                "test-product-translated-node-english"
            ]
        },
        "locale": "en"
    }
]
```
We can see that it work fine with translated product, but for untranslated blog post it returns inexistent path for "en" locale (second path in this example).
And the more locales you have, the more non-existent paths will be returned.

That triggers another problems with next-sitemap for example.

**Comment by shadcn** [comment]
> Maybe we can check node for system field "content_translations" somewhere in getPathsFromContext or buildPathsFromResources and return only such path which is exist in both nextjs and drupal locales?

We could. I'll look into it.

For now, I think what you could do is use two different calls for paths.

```ts
const pathsWithLocale = await getPathsFromContext("node--product", context)
const pathsWithoutLocale = await getPathsFromContext("node--blog", {})

const paths = [
    ...pathsWithLocale,
    ...pathsWIthoutLocale,
]
```

Can you give this a try and let me know?

**Comment by RyadRiad16** [comment]
hi , have a selimlar problem with the stater kit : 
in my drupal 10 ( varbase ) i have tow languages , defaut  /fr and /es .
the articles are not translated . every language has  it own articles .
in the starter kit the default language works fine but i get 404 in full es articles .
how can we deal with that ? is there any solution ?
